src/headers.ts

```diff
@@ -45,34 +47,37 @@ export enum CacheControl {
 
 export enum ContentType {
   APPLICATION_JAVASCRIPT = 'application/javascript',
-  APPLICATION_JSON = 'application/json; charset=UTF-8',
-  TEXT_JAVASCRIPT = 'text/javascript; charset=UTF-8',
-  TEXT_PLAIN = 'text/plain; charset=UTF-8',
+  APPLICATION_JSON = 'application/json',
+  TEXT_JAVASCRIPT = 'text/javascript',
+  TEXT_PLAIN = 'text/plain',
 }
 
 export enum ContentEncoding {
   BROTLI = 'br',
 }
 
 /**
- * Chooses the Content-Type of the output based on the input response's value.
+ * Normalizes the Content-Type of the output based on the input response's value.
  *
- * * Overrides `application/javascript` to `text/javascript; UTF-8`
- * * If defined, uses the input response's Content-Type directly,
+ * * Overrides `application/javascript` to `text/javascript; charset=UTF-8`
+ * * If defined, uses the input response's Content-Type directly (adding
+ *   `; charset=UTF-8` to known types if it is missing),
  * * Otherwise fallbacks back to `text/plain; charset=UTF-8`
  *
  * @param inputContentType - value of the input response's Content-Type header.
  */
-function chooseContentType(inputContentType: string | null): string {
+function normalizeContentType(inputContentType: string | null): string {
   if (inputContentType === ContentType.APPLICATION_JAVASCRIPT) {
-    return ContentType.TEXT_JAVASCRIPT;
+    return `${ContentType.TEXT_JAVASCRIPT}; ${CHARSET_UTF_8}`;
   }
 
   if (inputContentType) {
-    return inputContentType;
+    return Object.values(ContentType).map(String).includes(inputContentType)
```

Comment:
Nit: `.map(String)` shouldn't be necessary? Or, we could solve with a typecast instead.

Reply:
It's indeed a typing issue, without `.map(String)` I get `Argument of type 'string' is not assignable to parameter of type 'ContentType'`

I personally prefer the simplicity/explicitness of `.map(String)`, assuming we're not micro-optimizing the performance here. Casting is the last-resort choice for me because it kind of nullifies the point of strict typing. wdyt - do we really care about optimizing this?

Reply:
Up to you.

Reply:
I'd rather keep it this way then :)